Remove unusable C-linkage in source_sampling.h

If you try to build Pyne with MOOSE's conda environment (which uses the `-Wreturn-type-c-linkage` flag), you get an error due to unusable C-linkage:

```
/Users/vishalpatel/mambaforge3/envs/moose/include/pyne.h:6104:23: error: 'read_e_bounds' has C-linkage specified, but returns user-defined type 'std::vector<double>' which is incompatible with C [-Werror,-Wreturn-type-c-linkage]
  std::vector<double> read_e_bounds(std::string e_bounds_file);
```

`std::vector` is a C++ type, so it's not possible to compile functions that return this type with C. At least this is my understanding of what is happening.

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -10,6 +10,7 @@ Next Version
    * add composite actions to github workflows(#1459)
    * fixing typos in user warning message (#1456)
    * fix reorder error in member variables (#1466)
+   * remove unusable C-linkage with std::vector (#1468)
 
 v0.7.7
 ======

--- a/src/source_sampling.h
+++ b/src/source_sampling.h
@@ -42,10 +42,6 @@
 #endif
 #include "moab/CartVect.hpp"
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 namespace pyne {
 
   /// MCNP interface for source sampling setup
@@ -263,9 +259,5 @@ namespace pyne {
     bool check_cell_fracs(moab::Tag cell_fracs_tag);
   };
 } //end namespace pyne
-
-#ifdef __cplusplus
-} // extern "C"
-#endif
 
 #endif // PYNE_6OR6BJURKJHHTOFWXO2VMQM5EY

